prism/src/main/java/org/apache/falcon/jdbc/BacklogMetricStore.java

```diff
@@ -70,18 +70,19 @@ public synchronized void deleteMetricInstance(String entityName, String cluster,
         q.setParameter("clusterName", cluster);
         q.setParameter("nominalTime", nominalTime);
         q.setParameter("entityType", entityType.name());
-        try{
+        try {
             q.executeUpdate();
         } finally {
             commitAndCloseTransaction(entityManager);
         }
     }
 
-    public void deleteEntityInstance(String entityName){
+    public void deleteEntityInstance(String entityName, String entityType) {
         EntityManager entityManager = getEntityManager();
         beginTransaction(entityManager);
         Query q = entityManager.createNamedQuery(PersistenceConstants.DELETE_ALL_BACKLOG_ENTITY_INSTANCES);
         q.setParameter("entityName", entityName);
+        q.setParameter("entityType", entityType);
```

Comment:
but the query for the PersistenceConstants  DELETE_ALL_BACKLOG_ENTITY_INSTANCES expects three variables.
query = delete from BacklogMetricBean a where a.entityName = :entityName and a.clusterName = :clusterName and a.entityType = :entityType"
Which means from there you need to remove clusterName.